src/components/dialogs/dataSources/COVIDcast.svelte

```diff
@@ -23,38 +26,68 @@
     }
   }
 
-  onMount(() => {
-    fetchCOVIDcastMeta().then((res) => {
-      geoTypes = [...new Set(res.map((d) => d.geo_type))];
-      const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
-      for (const row of res) {
-        const ds = byDataSource.get(row.data_source);
-        if (!ds) {
-          byDataSource.set(row.data_source, {
-            label: row.data_source,
-            value: row.data_source,
-            signals: [row.signal],
-          });
-        } else if (!ds.signals.includes(row.signal)) {
-          ds.signals.push(row.signal);
+  function fetchMetadata() {
+    fetchCOVIDcastMeta((api_key = form_key)).then((res) => {
```

Comment:
Why not this format?
```suggestion
    fetchCOVIDcastMeta(form_key).then((res) => {
```